redfish/provider/resource_redfish_scp_export.go

```diff
@@ -0,0 +1,575 @@
+			Optional:            true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+		"share_name": schema.StringAttribute{
+			MarkdownDescription: `Share Name - The name of the directory or share on the server 
+			that contains the Server Configuration Profile file to export.`,
+			Description: `Share Name - The name of the directory or share on the server 
+			that contains the Server Configuration Profile file to export.`,
+			Optional: true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+		"share_type": schema.StringAttribute{
+			MarkdownDescription: "Share Type - The type of share being used to export the Server Configuration Profile file.",
+			Description:         "Share Type - The type of share being used to export the Server Configuration Profile file.",
+			Optional:            true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+				stringvalidator.OneOf([]string{
+					string("NFS"),
+					string("CIFS"),
+					string("HTTP"),
+					string("HTTPS"),
+					string("LOCAL"),
+				}...),
+			},
+		},
+
+		"target": schema.ListAttribute{
+			MarkdownDescription: "Filter configuration by target",
+			Description:         "Filter configuration by target",
+			Optional:            true,
+			ElementType:         types.StringType,
+			Computed:            true,
+			Default: listdefault.StaticValue(
+				types.ListValueMust(
+					types.StringType,
+					[]attr.Value{
+						types.StringValue("ALL"),
+					},
+				),
+			),
+			Validators: []validator.List{
+				listvalidator.SizeAtLeast(1),
+				listvalidator.ValueStringsAre(stringvalidator.OneOf([]string{
+					string("ALL"),
+					string("IDRAC"),
+					string("BIOS"),
+					string("NIC"),
+					string("RAID"),
+					string("FC"),
+					string("InfiniBand"),
+					string("SupportAssist"),
+					string("EventFilters"),
+					string("System"),
+					string("LifecycleController"),
+					string("AHCI"),
+					string("PCIeSSD"),
+				}...)),
+			},
+		},
+		"username": schema.StringAttribute{
+			MarkdownDescription: `Username - The username to use when authenticating with the server
+			 that contains the Server Configuration Profile file being exported.`,
+			Description: `Username - The username to use when authenticating with the server
+			 that contains the Server Configuration Profile file being exported.`,
+			Optional: true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+	}
+}
+
+// ValidateConfig validates the resource config.
+func (*ScpExportResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	// Get Plan Data
+	var plan models.TFRedfishScpExport
+	diags := req.Config.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	sp := plan.ShareParameters
+	shareType := sp.ShareType.ValueString()
```

Comment:
Shares are optional, so
```suggestion
	sp := plan.ShareParameters
	if sp.IsNull() || sp.IsUnknown() {
	        return
	}
	shareType := sp.ShareType.ValueString()
```

Reply:
Same for import resource

Reply:
Share parameter should be required. 
will change it to required